chore(repo): add taiko-client to taiko-mono

d1onys1us opened the pull request:
## summary of changes
This PR adds taiko-client into taiko-mono, so it can benefit from our CICD. There are two commits to review this PR:

- dd4a2712c5b20b54736f0952501ecc94384d317f is an **exact copy** of taiko-client at commit **15cf108**. you can verify this with something like `diff -r packages/taiko-client ../taiko-client` if you have taiko-mono and taiko-client in the same directory on your local machine.
- ca8cc17b4dc00140f43ce0659977baafd6259e36 this is where i make the necessary modifications to integrate taiko-client into taiko-mono. it is only removing taiko-mono directory dependency and fixing the CICD workflows.
- a1eb2ae6be461900476590db849549ae32babd67 this is where i simply do a find and replace for the go module paths.

## next steps
1. Let this merge into taiko-mono.
2. Add the `GAR_JSON_KEY` to taiko-mono for the docker job. <- @davidtaikocha
3. Rebase all PRs in taiko-client against this and open in taiko-mono, you'll just need to replace the workspace module path: `github.com/taikoxyz/taiko-client` -> `github.com/taikoxyz/taiko-mono/packages/taiko-client`.
4. Move any github issues from taiko-client to taiko-mono. <- @d1onys1us
5. Mark taiko-client as a public readonly archive, and remove it from the taikoxyz pinned repos. <- @dantaik

dantaik commented:
According to @davidtaikocha  there is one or two changes that needs to be merged into the codebase before merging this PR.

d1onys1us commented:
@dantaik @davidtaikocha i fixed up this PR and got it ready for review. you can check my description, i split the PR into three logically separated commits for easier review of the changes.
